### DISCUSSION TITLE
Memory limit

### DISCUSSION THREAD

**Discussion by Rattnik** [PrefectHQ/prefect#3945] [Q&A]
Hi, during running some tasks, my Virtual Mashine got fault with the error: `**Out of Memory: Killed proccess (prefect)**`
How should I limit max memory size which Prefect can use??
Now I have only 16GB RAM.

**Comment by joshmeek** [comment]
Hmm @Rattnik is this your prefect flow that is running or the server? I haven't experienced server ever running out of memory but I have seen cases where some users write flows that are very memory intensive.

**Reply by Rattnik** [reply]
@joshmeek Yes, it happens during flow running. But, can I limit maximum memory for all flows?

**Reply by joshmeek** [reply]
They only way to currently limit memory for flow runs would be using one of the [RunConfig](https://docs.prefect.io/orchestration/flow_config/run_configs.html) options such as a KubernetesRun and limiting the amount of memory on the run there. Outside of that you would have to build something custom.

**Reply by Rattnik** [reply]
Ok. Thanks.
But Kubernetes just kill process which use memory greater than limit.
